pkg/dmaasbackup/schedule.go

```diff
@@ -72,12 +77,84 @@ func (d *dmaasBackup) generateBackupName(dbkp v1alpha1.DMaaSBackup) string {
 // updateScheduleInfo checks for relevant velero schedule/backup and
 // update the dbkp with schedule/backup details
 func (d *dmaasBackup) updateScheduleInfo(dbkp *v1alpha1.DMaaSBackup) error {
-	// TODO
+	var (
+		schedule *velerov1api.Schedule
+		bkp      *velerov1api.Backup
+		err      error
+	)
+
+	// check if dbkp is having config for non-scheduled backup
+	if yes := isSchedule(*dbkp); !yes {
+		bkp, err = d.backupClient.Get(dbkp.Name, metav1.GetOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return errors.Wrapf(err, "failed to update scheduleInfo")
+		}
+		if bkp.Name == "" {
+			// Backup could have deleted due to expiration of TTL
+			// TODO: Should we clear backupname from dmaasbackup? and/or recreate?
+			return nil
+		}
+		if dbkp.Status.VeleroBackupName == nil {
+			backupName := bkp.Name
+			dbkp.Status.VeleroBackupName = &backupName
+		}
+		return nil
+	}
+
+	// check for any queued dummy entry in veleroschedule
+	dummySchedule := getDummyVeleroSchedule(dbkp)
```

Comment:
why we need this code of checking for empty Schedule and updating it if exists?
This should get handled in processNonperiodicConfigSchedule and processPeriodicConfigSchedule functions by checking for error of AlreadyExists